sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/SynchronousMessageSubscriber.java

```diff
@@ -236,6 +236,11 @@ private void drainQueue() {
                 numberRequested = REQUESTED.addAndGet(this, -numberConsumed);
             }
         }
+        if (numberRequested == 0L) {
```

Comment:
This seems to be the successfully complete current work case?
Where does code go if current work timeout or killed?

Reply:
If current work timeout, as you could see in the diagram I provided, the current work will be updated in the line: [SynchronousMessageSubscriber.java#L202](https://github.com/Azure/azure-sdk-for-java/blob/982b06d7163536c5aefc15be641ba52e9cbdf2fe/sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/SynchronousMessageSubscriber.java#L202)

If receive thread is killed, the work won't be killed because it is scheduled on reactor thread. So the work keep receiving message till complete. But there is not subscriber on that work anymore, this is another issue currently I am looking at,  how to tell the current work if the receive thread is killed.